app/models/subcase.js

```diff
@@ -144,25 +144,26 @@ export default ModelWithModifier.extend({
     return null;
   }),
 
-  approved: computed('treatments', 'treatments.@each.decisionResultCode', function() {
-    return PromiseObject.create({
-      promise: this.get('treatments').then((treatments) => {
-        if (treatments && treatments.get('length') > 0) {
-          const treatmentIds = treatments.map((treatment) => treatment.get('id')).join(',');
-          const drcIds = ['56312c4b-9d2a-4735-b0b1-2ff14bb524fd', '9f342a88-9485-4a83-87d9-245ed4b504bf'].join(',');
-          return this.store.query('agenda-item-treatment', {
-            filter: {
-              id: treatmentIds,
-              'decision-result-code': {
-                id: drcIds,
-              },
+  approved: computed('treatments', 'treatments.@each.decisionResultCode', async function() {
+    const meeting = await this.get('requestedForMeeting');
+    if (meeting.isFinal) {
+      const treatments = await this.get('treatments');
+      if (treatments && treatments.get('length') > 0) {
+        const treatmentIds = treatments.map((treatment) => treatment.get('id')).join(',');
+        const drcIds = ['56312c4b-9d2a-4735-b0b1-2ff14bb524fd', '9f342a88-9485-4a83-87d9-245ed4b504bf'].join(',');
```

Comment:
Zou je deze nog kunnen refactoren om van hier https://github.com/kanselarij-vlaanderen/kaleidos-frontend/blob/21526319b6e605b5691cdc489962010c72842c0a/app/utils/config.js#L193 gesourced te worden? (je kan in de `queryOne` dan filteren als `decision-result-code[:uri:]: ...`.

Reply:
lokaal geprobeerd:
```
const approvedTreatment = await this.store.queryOne('agenda-item-treatment', {
          'filter[id]': treatmentIds,
          'filter[decision-result-code][:uri:]': drcUris,
          include: 'decision-result-code',
        });
```
Maar dit werkt niet, ik zou 1 treatment moeten krijgen maar krijg er geen.
De query werkt wel als ik enkel de uri van "goedgekeurd" erin zet.
Kan deze filter niet om met meerdere waarden ?

Reply:
Ik heb zonet exact hetzelfde voor. :thinking: Ik ga't navragen.

Reply:
Zie https://chat.semte.ch/channel/mu-semtech?msg=QDSMw3g9jSXsviyhq voor context omtrent het ophalen van codelijsten via URI. Hier ms idd buiten scope. Ik probeer er nu en dan tussendoor 1 op te nemen en om te zetten naar URI-only ... deze zal later aan beurt komen.